packages/@react-aria/landmark/src/useLandmark.ts

```diff
@@ -0,0 +1,208 @@
+        nextLandmark = walker.lastChild();
+        let temp = nextLandmark;
+        while (temp) {
+          nextLandmark = temp;
+          walker.currentNode = temp;
+          temp = walker.lastChild();
+        }
+      }
+    } else {
+      nextLandmark = walker.nextNode();
+      if (!nextLandmark) {
+        walker.currentNode = document.body;
+        nextLandmark = walker.firstChild();
+      }
+    }
+    return nextLandmark;
+  }
+
+  // skipping nodes in the current landmark, look for the next landmark, wrap if needed
+  // if that landmark had something focused in it previously, focus that
+  // else focus the first focusable element
+  // hopefully getFocusableTreeWalker will do the searching, or some derivative of it
+  // can we use FocusScope to handle the refocusing inside the landmark?
+  // essentially treat landmarks as a focus scope but without wrapping or restore or containment
+  public f6Handler(e: KeyboardEvent) {
+    if (e.key === 'F6') {
+      e.preventDefault();
+      e.stopPropagation();
+
+      let currentLandmark = this.closestLandmark(e.target as HTMLElement)?.current;
+      let walker = getLandmarkTreeWalker(
+        document.body,
+        {
+          from: currentLandmark || document.body,
+          registeredLandmarks: Array.from(this.landmarks.keys()).map(ref => ref.current)
+        }
+      );
+      let nextLandmark = this.getNextLandmark(walker, {backward: e.shiftKey});
+
+      if (nextLandmark) {
+        // if something was previously focused in this landmark, then return focus to it
+        let key = Array.from(this.landmarks.keys()).find(ref => ref.current === nextLandmark);
+        if (key && this.landmarks.has(key)) {
+          let lastFocused = this.landmarks.get(key);
+          if (document.body.contains(lastFocused)) {
+            lastFocused.focus();
+            return;
+          }
+        }
+        // otherwise find an element to focus
+        let focusableWalker = getFocusableTreeWalker(nextLandmark);
+        let nextElement = focusableWalker.nextNode() as HTMLElement;
+        if (!nextElement) {
+          walker.currentNode = nextLandmark;
+          nextElement = focusableWalker.firstChild() as HTMLElement;
+        }
+        while (this.closestLandmark(nextElement)?.current !== nextLandmark) {
+          nextElement = focusableWalker.nextNode() as HTMLElement;
+        }
+        if (nextElement) {
+          nextElement.focus();
+        }
+      }
+    }
+  }
+
+  // Only the closest landmark cares about focus inside of it
+  public focusinHandler(e: FocusEvent) {
+    let currentLandmark = this.closestLandmark(e.target as HTMLElement);
+    this.landmarks.set(currentLandmark, e.target as HTMLElement);
+  }
+}
+
+document.addEventListener('keydown', LandmarkManager.getInstance().f6Handler.bind(LandmarkManager.getInstance()), {capture: true});
```

Comment:
these should be setup once but only after useLandmark is invoked

Reply:
Can we use `useGlobalListener` here?

Reply:
made a `setup()` method that is called right after initialization. Should add once, after useLandmark is invoked. Let me know if there is a better method.

Reply:
And I wonder if we should remove those listeners if a landmark is removed and there are no more landmarks in the manager?

Reply:
yeah I think that would be reasonable